src/input/reader_to_storage.cc

```diff
@@ -194,83 +194,98 @@ StorageBase * ReaderToStorage::make_storage(PathBase &p, const Type::TypeBase *t
 StorageBase * ReaderToStorage::make_storage(PathBase &p, const Type::Record *record)
 {
 	// control test, check correct tag (or TYPE key) if Record is derived from Abstract
-	string record_name_from_tag = p.get_record_name();
-	if ( record_name_from_tag != "" ) {
-		ASSERT(record_name_from_tag == record->type_name())(record_name_from_tag)(record->type_name()).error("Inconsistent tag of record.");
+	PathBase::RecordTagType tag_type;
+	string record_name_from_tag;
+	p.get_record_tag(tag_type, record_name_from_tag);
```

Comment:
Rather keep original bahavior of type_name, just rename it to get_tag, and check "include" and include_csv" here.